vitrivr-engine-index/src/main/kotlin/org/vitrivr/engine/index/transform/TemplateTextTransformer.kt

```diff
@@ -0,0 +1,73 @@
+package org.vitrivr.engine.index.transform
+
+import io.github.oshai.kotlinlogging.KotlinLogging
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
+import org.vitrivr.engine.core.context.Context
+import org.vitrivr.engine.core.context.IndexContext
+import org.vitrivr.engine.core.model.content.ContentType
+import org.vitrivr.engine.core.model.content.factory.ContentFactory
+import org.vitrivr.engine.core.model.retrievable.Retrievable
+import org.vitrivr.engine.core.model.retrievable.attributes.ContentAuthorAttribute
+import org.vitrivr.engine.core.operators.Operator
+import org.vitrivr.engine.core.operators.general.Transformer
+import org.vitrivr.engine.core.operators.general.TransformerFactory
+
+private val logger = KotlinLogging.logger {}
+
+/**
+ * A [Transformer] that takes an input template with placeholders and inserts content from fields in their place.
+ *
+ * @author Laura Rettig
+ * @version 1.0.0
+ */
+class TemplateTextTransformer : TransformerFactory {
+    override fun newTransformer(name: String, input: Operator<out Retrievable>, context: Context): Transformer {
+        val template = context[name, "template"] ?: throw IllegalArgumentException("The template text transformer requires a template.")
+        val regex = "\\$\\{([^}]+)\\}".toRegex()
+        val contentFields = regex.findAll(template).map { it.groupValues[1] }.toList()
+        val defaultValue = context[name, "defaultValue"] ?: ""
```

Comment:
magic values should be defined as constants

Reply:
Especially if they're RegEx, which incur a cost for compilation.